src/test/web/index.ts

```diff
@@ -47,7 +50,9 @@ export async function activate(context: IExtensionContext): Promise<IExtensionAp
 
             try {
                 // Run the mocha test
-                mocha.run((failures) => {
+                mocha.run(async (failures) => {
+                    // Wait for a while, till we've managed to send the test results to the test server.
+                    await sleep(5_000);
```

Comment:
We only need a few ms, but added 5s, basically we cannot kill VS Code web until all the results have been sent through. I didn't want to complicate the code by adding more checks & locks to determine whether all the data has gone through, this should work 99.9% of the time & keeps the code simple & clean.